src/toxcore/dht/server/mod.rs

```diff
@@ -570,28 +569,33 @@ impl Server {
         }
     }
 
-    /// Send UDP packet to specified address. The packet will be dropped if IPv6
-    /// mode is disabled but the address is IPv6 address.
-    fn send_to(&self, addr: SocketAddr, packet: DhtPacket) -> IoFuture<()> {
-        if self.is_ipv6_mode {
-            match addr.ip() {
-                IpAddr::V4(ip) => {
-                    let ip_v6 = ip.to_ipv6_mapped();
-                    let addr = SocketAddr::new(IpAddr::V6(ip_v6), addr.port());
-                    send_to(&self.tx, (packet, addr))
-                },
-                IpAddr::V6(_ip) => {
-                    send_to(&self.tx, (packet, addr))
-                },
-            }
-        } else {
-            if addr.is_ipv4() {
-                send_to(&self.tx, (packet, addr))
-            } else {
-                debug!("Attempting to send packet to IPv6 address wile DHT node is running in IPv4 mode");
-                Box::new(future::ok(()))
-            }
+    /// Send UDP packet node. If the node has both IPv4 and IPv6 addresses,
+    /// then it sends packet to both addresses.
+    fn send_to(&self, node: &DhtNode, packet: DhtPacket) -> IoFuture<()> {
+        let mut futures = Vec::new();
+
+        if self.is_ipv6_mode {// DHT node is running in ipv6 mode
+            if let Some(sock_v6) = node.saddr_v6 {
+                futures.push(send_to(&self.tx, (packet.clone(), SocketAddr::V6(sock_v6))));
+            };
+
+            if let Some(sock_v4) = node.saddr_v4 {
+                let ip_v6 = sock_v4.ip().to_ipv6_mapped();
+                let addr = SocketAddr::new(IpAddr::V6(ip_v6), sock_v4.port());
+                futures.push(send_to(&self.tx, (packet, addr)));
+            };
+        } else { // DHT node is running in ipv4 mode
+            if let Some(sock_v4) = node.saddr_v4 {
+                futures.push(send_to(&self.tx, (packet, SocketAddr::V4(sock_v4))));
+            };
         }
+
```

Comment:
Add warning if `futures` are empty, something like: `Trying to send a packet to node PK but could not find appropriate addr`